gusto/terms.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABCMeta, abstractmethod
+from firedrake import FacetNormal, div, dx, inner
+
+
+class Term(object, metaclass=ABCMeta):
+
+    off_centering = 0.5
+
+    def __init__(self, state, test):
+        self.state = state
+        self.test = test
+        self.parameters = state.parameters
+        self.n = FacetNormal(state.mesh)
```

Comment:
Why does this guy need to carry around a symbolic facet normal? Why can't you make it when you need it?